src/mlpack/tests/main_tests/range_search_test.cpp

```diff
@@ -0,0 +1,389 @@
+/**
+ * @file range_search_test.cpp
+ * @author Niteya Shah
+ *
+ * Test mlpackMain() of range_search_main.cpp.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include <string>
+
+#define BINDING_TYPE BINDING_TYPE_TEST
+static const std::string testName = "RangeSearchMain";
+
+#include <mlpack/core.hpp>
+#include <mlpack/core/util/mlpack_main.hpp>
+#include "test_helper.hpp"
+#include <mlpack/methods/range_search/range_search_main.cpp>
+
+#include <boost/test/unit_test.hpp>
+#include "../test_tools.hpp"
+
+using namespace mlpack;
+
+struct RangeSearchTestFixture
+{
+ public:
+
+  RangeSearchTestFixture()
+  {
+    // Cache in the options for this program.
+    CLI::RestoreSettings(testName);
+  }
+  ~RangeSearchTestFixture()
+  {
+    // Clear the settings.
+    bindings::tests::CleanMemory();
+    CLI::ClearSettings();
+  }
+};
+
+BOOST_FIXTURE_TEST_SUITE(RangeSearchMainTest, RangeSearchTestFixture);
+/*
+* Check that the correct output is returned for a small synthetic input
+* case
+*/
+BOOST_AUTO_TEST_CASE(RangeSearchTest)
+{
+  //Matrix Input is expected in this format
+  arma::mat x = {{0, 3, 3, 4, 3, 1},
+                 {4, 4, 4, 5, 5, 2},
+                 {0, 1, 2, 2, 3, 3}};
+  std::string distancefile = "distances.csv";
+  std::string neighborsfile = "neighbors.csv";
+  double minv = 0, maxv = 3;
+  vector<vector<size_t>> neighborval = {{},
+                                        {2, 3, 4},
+                                        {1, 3, 4, 5},
+                                        {1, 2, 4},
+                                        {1, 2, 3},
+                                        {2}};
+  vector<vector<double>> distanceval = {{},{1, 1.73205, 2.23607},
+                                        {1, 1.41421, 1.41421, 3},
+                                        {1.73205, 1.41421, 1.41421},
+                                        {2.23607, 1.41421, 1.41421},
+                                        {3}};
+  vector<vector<size_t>> neighbors;
+  vector<vector<double>> distances;
+  SetInputParam("reference", move(x));
+  //To prevent warning for lack of definition
+  SetInputParam("min", minv);
+  SetInputParam("max", maxv);
+  SetInputParam("distances_file", distancefile);
+  SetInputParam("neighbors_file", neighborsfile);
+
+  mlpackMain();
+
+  neighbors = ReadData<size_t>(neighborsfile);
+  distances = ReadData<double>(distancefile);
+
+  CheckMatrices(neighbors, neighborval, 1e-5f);
+  CheckMatrices(distances, distanceval);
+}
+//Perform a Test for Search with a Query
```

Comment:
Please put an empty line between the end of the previous test and the beginning of this one.  And a more comprehensive comment describing this test would be nice also.